astropy/time/formats.py

```diff
@@ -573,6 +573,69 @@ def value(self):
 
         return iterator.operands[-1]
 
+    def to_value(self, timezone=None):
+        """
+        Convert to (potentially timezone-aware) `~datetime.datetime` object.
+
+        If ``timezone`` is not `None`, return a timezone-aware datetime
+        object.
+
+        Parameters
+        ----------
+        timezone : {`~datetime.tzinfo`, None} (optional)
+            If not `None`, return timezone-aware datetime.
+
+        Returns
+        -------
+        `~datetime.datetime`
+            If ``timezone`` is not `None`, output will be timezone-aware.
+        """
+        if timezone is not None:
+            if self._scale != 'utc':
+                raise ScaleValueError("Scale is {}, must be UTC."
+                                      "".format(self._scale))
+
+        # Below is mostly copied from self.value, with added timezone handling
```

Comment:
A year from now, this will be a rather unclear comment, so I would suggest just removing it. Instead, at the definition of `to_value`, you might want to add something along the lines of "Rather than define a `value` property directly, we have a function, since we want to be able to pass in timezone information."